adapter/internal/discovery/xds/server.go

```diff
@@ -346,11 +346,15 @@ func DeleteAPIs(vhost, apiName, version string, environments []string) error {
 		return err
 	}
 
-	// if this is the final vhost delete map entry
-	if _, ok := vhosts[vhost]; len(vhosts) == 1 && ok {
-		logger.LoggerXds.Debugf("There are no vhosts for the API %v and clean vhost entry with name:version from maps",
-			apiNameVersionID)
-		delete(apiToVhostsMap, apiNameVersionID)
+	if _, ok := vhosts[vhost]; ok {
+		if len(vhosts) == 1 {
+			// if this is the final vhost delete map entry
+			logger.LoggerXds.Debugf("There are no vhosts for the API %v and clean vhost entry with name:version from maps",
```

Comment:
This log message is confusing :)

Reply:
fixed with https://github.com/wso2/product-microgateway/pull/1949/commits/de3529307cac69e02b7088edd6c868fe7ab356b7